src/app/graph-service.ts

```diff
@@ -1,71 +1,75 @@
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
 // ------------------------------------------------------------------------------
 
 import { Injectable } from '@angular/core';
-import { Headers, Http, Response, ResponseContentType } from '@angular/http';
-
-import { AllowedGraphDomains, RequestType } from './base';
+import { Http, Response, ResponseContentType, Headers } from '@angular/http';
+import { AuthService } from './authentication/auth.service';
+import { RequestType, AllowedGraphDomains } from './base';
+import 'rxjs/add/operator/toPromise';
 
 @Injectable()
 export class GraphService {
-  constructor(private http: Http) { }
+    constructor(private http: Http, private authService: AuthService) { }
 
-  public performAnonymousQuery(queryType: RequestType, query: string, headers?: Headers): Promise<Response> {
+    performAnonymousQuery(queryType:RequestType, query:string, headers?:Headers):Promise<Response> {
         if (!headers) {
             headers = new Headers();
         }
         headers.append('Authorization', 'Bearer {token:https://graph.microsoft.com/}');
 
         if (queryType === 'GET') {
-            return this.http.get(`https://proxy.apisandbox.msdn.microsoft.com/svc?url=${encodeURIComponent(query)}`,
-              {headers}).toPromise();
+            return this.http.get(`https://proxy.apisandbox.msdn.microsoft.com/svc?url=${encodeURIComponent(query)}`, {headers}).toPromise();
         } else if (queryType === 'GET_BINARY') {
-            return this.http.get(`https://proxy.apisandbox.msdn.microsoft.com/svc?url=${encodeURIComponent(query)}`,
-              {headers, responseType: ResponseContentType.ArrayBuffer}).toPromise();
+            return this.http.get(`https://proxy.apisandbox.msdn.microsoft.com/svc?url=${encodeURIComponent(query)}`, {headers, responseType: ResponseContentType.ArrayBuffer}).toPromise();
         }
     }
+    // let method = isAuthenticated() ? this.GraphService.performQuery : this.GraphService.performAnonymousQuery;
 
-  public performQuery = (queryType: RequestType, query: string, postBody?: any, requestHeaders?: Headers) => {
-        // Make sure the request is being sent to the Graph and not another domain
+    performQuery = (queryType:RequestType, query:string, postBody?:any, requestHeaders?:Headers) => {
+        // make sure the request is being sent to the Graph and not another domain
         let sentToGraph = false;
 
-        for (const domain of AllowedGraphDomains) {
+        for (let domain of AllowedGraphDomains) {
             if (query.startsWith(domain)) {
                 sentToGraph = true;
                 break;
             }
         }
 
         if (!sentToGraph) {
-            throw new Error('Not sending request to known Graph deployment');
+            throw 'Not sending request to known Graph deployment';
```

Comment:
We want the message wrapped in an Error object. Especially once we start extending the Error object, it will give more information about the error source for debugging. If we need to provide additional context to a error handler, we should name the error.